appyx-components/modal/common/src/commonMain/kotlin/com/bumble/appyx/components/modal/ui/ModalMotionController.kt

```diff
@@ -0,0 +1,101 @@
+package com.bumble.appyx.components.modal.ui
+
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.unit.Density
+import androidx.compose.ui.unit.DpOffset
+import androidx.compose.ui.unit.dp
+import com.bumble.appyx.components.modal.ModalModel
+import com.bumble.appyx.components.modal.operation.FullScreen
+import com.bumble.appyx.components.modal.operation.Revert
+import com.bumble.appyx.components.modal.operation.Show
+import com.bumble.appyx.interactions.core.ui.context.TransitionBounds
+import com.bumble.appyx.interactions.core.ui.context.UiContext
+import com.bumble.appyx.interactions.core.ui.gesture.*
+import com.bumble.appyx.interactions.core.ui.property.impl.*
+import com.bumble.appyx.interactions.core.ui.state.MatchedTargetUiState
+import com.bumble.appyx.transitionmodel.BaseMotionController
+
+class ModalMotionController<InteractionTarget : Any>(
+    uiContext: UiContext,
+) : BaseMotionController<InteractionTarget, ModalModel.State<InteractionTarget>, MutableUiState, TargetUiState>(
+    uiContext = uiContext
+) {
+    private val width = uiContext.transitionBounds.widthDp
+    private val height = uiContext.transitionBounds.heightDp
+
+    private val createdState: TargetUiState =
+        TargetUiState(
+            width = Width.Target(0f),
+            height = Height.Target(0f),
+            position = Position.Target(DpOffset(0.dp, 0.dp)),
+            corner = BackgroundCorner.Target(40f),
+        )
+    private val modalState: TargetUiState =
+        TargetUiState(
+            width = Width.Target(0.7f),
+            height = Height.Target(0.5f),
+            position = Position.Target(DpOffset(0.dp, height * 0.5f)),
+            corner = BackgroundCorner.Target(40f),
+        )
+
+    private val fullScreenState: TargetUiState =
+        TargetUiState(
+            width = Width.Target(1f),
+            height = Height.Target(1f),
+            position = Position.Target(DpOffset(0.dp, 0.dp)),
+            corner = BackgroundCorner.Target(0.1f),
+        )
+    
+    private val destroyedState: TargetUiState =
+        TargetUiState(
+            width = Width.Target(1f),
+            height = Height.Target(1f),
+            position = Position.Target(DpOffset(width, 0.dp)),
```

Comment:
I'd rather not animate width. Let's try to replicate standard visual expectations of modals (only vertical size changes or movement).

Reply:
I don't have a strong opinion about that but in the appyx 1.0 example it was changing both width and height (let me know if I misunderstood things, I easily might have)

Reply:
If that's the case than sorry, probably the 1.0 example was doing it wrong too!

`Width` can stay, no issue with it.